src/ng-generate/components/shared/generators/components/command-bar/files/__name@dasherize__-command-bar.component.ts.template

```diff
@@ -107,4 +124,54 @@ export class <%= classify(name) %>CommandBarComponent {
             this.customCommandBarActionEvent.emit({action: action})
         }
     <% } %>
+
+    ngAfterViewInit(): void {
+      const compWidths: any[] = this.components
+      .toArray()
+      .map((comp) => 
+        ( {width: comp.nativeElement.clientWidth, id: comp.nativeElement.id }));
+
+      if(this.initialCompWidths.length === 0){
+        this.initialCompWidths = compWidths;
+      };
+
+      this.updateToolbarComponentsVisibility();
+    }
+
+    @HostListener('window:resize', ['$event'])
+    onResize() {
+       this.updateToolbarComponentsVisibility();
+    }
+
+  updateToolbarComponentsVisibility() {
+      this.hiddenComponents = [];
+      const containerWidth = this.componentsContainer.nativeElement.clientWidth;
+      let usedWidth = 0;
+
+       for (let i = 0; i < this.initialCompWidths.length; i++) {
+          const buttonWidth = this.initialCompWidths[i];
+          // 70 is for the width of the collapsed menu width
+          if (usedWidth + buttonWidth.width + 70 >= containerWidth) {
+            this.hiddenComponents.push(buttonWidth.id);
+          } else {
+            usedWidth += buttonWidth.width;
+          }
+        }
+    }
+
+    isInCollapsedMenu(id: string) {
+      return this.hiddenComponents.includes(id);
+    }
+
+    <% if (options.componentType === 'table' && options.hasSearchBar) { %>
+      openSettingsFromCollapsed() {      
+        const configMenuDialogRef =  this.dialog.open(MovementTableConfigMenuComponent, {
```

Comment:
the dialog component should be model-specific here 
```suggestion
        const configMenuDialogRef =  this.dialog.open(<%= classify(name) %>TableConfigMenuComponent, {
```

Reply:
done